### PR TITLE
Use variable name for native question field filters when mapping

### LINKED ISSUE #13376: 0.37-rc1: When mapping dashboard filters to columns, SQL questions should display the name of the column mapped to the field filter
Right now, we display the name of the field filter variable, which doesn't seem as helpful as displaying the mapped table and column name.

### LINKED ISSUE #13406: 0.37.0-rc1: Variables from Saved Question are referencing the same question
When there's multiple nested questions on a dashboard, and linking the variable, then it says it's the same original question referenced - even though it's 3 different completely nested questions (and Saved Question) in this example:
image

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Fixes #13376, Fixes #13406

@flamber how does this look?

dashboard-parameter-mapper

### CODE REVIEW

**Comment by flamber** [comment]
I like that :+1:, but probably want others to okay it too.

File: frontend/src/metabase/dashboard/containers/DashCardCardParameterMapper.jsx [hunk 1]
```diff
@@ -172,6 +172,8 @@ export default class DashCardCardParameterMapper extends Component {
                 <span className="text-centered mr1">
                   {disabled
                     ? t`No valid fields`
+                    : selected && selected.sectionName == null
+                    ? selected.name // for native question variables we just display the name
```
Comment (kdoh) [inline frontend/src/metabase/dashboard/containers/DashCardCardParameterMapper.jsx:176]: This ternary feels like it's getting a bit out of hand.

Comment (paulrosenzweig) [reply frontend/src/metabase/dashboard/containers/DashCardCardParameterMapper.jsx:176]: Good point. I guess it's doing two things: 1. Are we showing a name? 2. What name are we showing?

I'll extract the second one to elsewhere.

**Review by robdaemon — APPROVED** [review]
looks good to me, code-wise, if someone else wants to comment on the UX

**Comment by paulrosenzweig** [comment]
> if someone else wants to comment on the UX

Maz and I chatted about this UX, but I didn't put that anywhere besides Slack. My takeaway was that there's some question about whether we'll want to show table/field names for native question field filter rather than the variable names as we do now. We're good shipping this and reassessing down the line.

**Pull request merged**